frontstage/controllers/oauth_controller.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from flask import current_app as app
+import requests
+from structlog import wrap_logger
+
+from frontstage.exceptions.exceptions import ApiError, OAuth2Error
+
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+def check_account_valid(username):
+    logger.debug('Attempting to check if account is valid in OAuth2 service')
+
+    url = f"{app.config['OAUTH_SERVICE_URL']}/api/v1/tokens/"
+    data = {
+        'grant_type': 'reset_password',
+        'client_id': app.config['OAUTH_CLIENT_ID'],
+        'client_secret': app.config['OAUTH_CLIENT_SECRET'],
+        'username': username,
+    }
+    headers = {
+        'Accept': 'application/json',
+        'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
+    }
+    response = requests.post(url, headers=headers, auth=app.config['OAUTH_BASIC_AUTH'], data=data)
+
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError:
+        if response.status_code == 401:
+            oauth2_error = response.json().get('detail')
+            logger.exception('Authentication error in OAuth2 service', error=oauth2_error)
```

Comment:
logger.exception logs at Error level not sure we want to be logging at Error level for every HTTPError caught. All these logger.exception logs will make our splunk alerts explode.

Reply:
@tej99 warning instead?

Reply:
@tej99 I was following what was implemented in response-operations, like [this for example](https://github.com/ONSdigital/response-operations-ui/blob/12946cb861e69a8285b98fecbf490858953b1d2c/response_operations_ui/controllers/respondent_controllers.py#L27). 

It's probably worth raising with the tech leads / @ONSdigital/sdc-ras to make sure we're not filtering legitimate errors to avoid noise at a Splunk level.

Reply:
It works fine in the example you were following because it is calling backstage-api which filters out errors, [like the 404](https://github.com/ONSdigital/ras-backstage/blob/da11ef32925c8ca924f977984ecd1da075bf93f8/ras_backstage/controllers/party_controller.py#L105) that should not be logging at error level.